api/services/User/index.js

```diff
@@ -37,44 +38,47 @@ export default class User extends BaseModelService {
     }
   }
 
-  static async findAvailaibilityDuring(
-    user_id,
-    dateTimeRange,
-    exculdeEvents = [],
-  ) {
-    const data = await Models.CalendarEventInvite.scope('visible').findOne({
-      where: {
-        user_id,
-      },
-      include: [
-        {
-          model: Models.CalendarEvent,
-          as: 'event',
-          where: {
-            [Models.Sequelize.Op.and]: [
-              overlapDateTimeClause(dateTimeRange),
-              {
-                id: {
-                  [Models.Sequelize.Op.notIn]: exculdeEvents,
+  static async findAllAvailabilityDuring(keys) {
+    const resultPromises = keys.map((key) => {
+      return Models.CalendarEventInvite.scope('visible').findOne({
+        where: {
+          user_id: key.user_id,
+        },
+        include: [
+          {
+            model: Models.CalendarEvent,
+            as: 'event',
+            where: {
+              [Models.Sequelize.Op.and]: [
+                overlapDateTimeClause(key.dateTimeRange),
+                {
+                  id: {
+                    [Models.Sequelize.Op.notIn]: key.exculdeEvents,
+                  },
                 },
-              },
-            ],
+              ],
+            },
+            required: true,
           },
-          required: true,
-        },
-      ],
+        ],
+      });
     });
+    const result = await Promise.all(resultPromises);
+    return result.map((data) => !data);
+  }
 
-    return !data;
+  static getAvailabilityLoader() {
+    return new DataLoader(this.findAllAvailabilityDuring);
   }
 
   @requireInstance
   ifAvailableDuring(dateTimeRange, exculdeEvents) {
-    return User.findAvailaibilityDuring(
-      this.instance.id,
+    const key = {
+      user_id: this.instance.id,
       dateTimeRange,
       exculdeEvents,
-    );
+    };
+    return User.getAvailabilityLoader().load(key);
```

Comment:
I am not sure about this. Is this alright? And where is `IfAvailableDuring()` used? I could not find it.

Reply:
Not being used anywhere right now.